File: 30-10-2024_Pairs_with_difference_k.py
class Solution:
    def countPairsWithDiffK(self, arr, k):
        maxi = 0
        size = len(arr)
        count = 0
        for i in range(size):
            if arr[i] > maxi:
                maxi = arr[i]
        
        ht = [0] * (maxi + 1)
        for i in range(size):
            ht[arr[i]] += 1
        
        for i in range(size):
            curr = arr[i]
            if curr + k <= maxi and ht[curr + k]:
                count += ht[curr + k]
        
        return count

File: 30-10-2024_Pairs_with_difference_k.java
class Solution {
    public int countPairsWithDiffK(int[] arr, int k) {
        int maxi = 0;
        int size = arr.length, count = 0;
        for (int i = 0; i < size; i++) {
            if (arr[i] > maxi)
                maxi = arr[i];
        }
        int[] ht = new int[maxi + 1];
        for (int i = 0; i < size; i++) ht[arr[i]]++;
        for (int i = 0; i < size; i++) {
            int curr = arr[i];
            if ((curr + k <= maxi) && ht[curr + k]>0) {
                count += ht[curr + k];
            }
        }
        return count;
    }
}
